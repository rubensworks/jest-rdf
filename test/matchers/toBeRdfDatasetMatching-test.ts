import { defaultGraph, literal, namedNode, quad } from '@rdfjs/data-model';
import datasetFactory = require('rdf-dataset-indexed');
import '../../index';

const quad1 = quad(
  namedNode('s'),
  namedNode('p'),
  namedNode('o'),
  namedNode('g'),
);

const quad2 = quad(
  namedNode('s'),
  namedNode('p'),
  namedNode('o'),
  defaultGraph(),
);

const quad3 = quad(
  namedNode('s'),
  namedNode('p'),
  literal('o'),
  defaultGraph(),
);

describe('#toBeRdfDatasetMatching', () => {
  it('should succeed if the dataset matches', () => {
    return expect(datasetFactory([quad1, quad2, quad3])).toBeRdfDatasetMatching({ subject: namedNode('s') }, 3);
  });

  it('should not succeed if the dataset does not match', () => {
    return expect(datasetFactory([quad1, quad2, quad3])).not.toBeRdfDatasetMatching({ subject: namedNode('s') });
  });

  it('should fail if the dataset does not match', () => {
    return expect(() => expect([
      expect(datasetFactory([quad1, quad2, quad3])).toBeRdfDatasetMatching({ subject: namedNode('s') }),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not fail if the dataset matches', () => {
    return expect(() => expect([
      expect(datasetFactory([quad1, quad2, quad3])).not.toBeRdfDatasetMatching({ subject: namedNode('s') }, 3),
    ])).toThrowErrorMatchingSnapshot();
  });
});
