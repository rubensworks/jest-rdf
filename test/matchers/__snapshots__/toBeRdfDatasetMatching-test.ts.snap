// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toBeRdfDatasetMatching should fail if the dataset does not match 1`] = `"expected dataset to match {\\"subject\\":{\\"value\\":\\"s\\"}} once"`;

exports[`#toBeRdfDatasetMatching should not fail if the dataset matches 1`] = `"expected dataset to not match {\\"subject\\":{\\"value\\":\\"s\\"}} 3 times"`;
